core/src/jsMain/kotlin/SharedRepeatableTask.kt

```diff
@@ -0,0 +1,46 @@
+package com.juul.kable
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.CoroutineStart.DEFAULT
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
+
+/**
+ * A mechanism for launching and awaiting some shared job repeatedly.
+ *
+ * The job is launched by calling [launchAsync]. Subsequent calls to [launchAsync] will return the
+ * same (i.e. shared) job if it is still executing.
+ */
+internal class SharedRepeatableTask(
+    private val scope: CoroutineScope,
+    private val job: suspend () -> Unit
+) {
+
+    private var deferred: Deferred<Unit>? = null
```

Comment:
A `Deferred` with a return type of `Unit` feels like it should just be a `Job`? (i.e. use `launch` instead of `async`)

It would also help remove the warning at the usage site:

![Screen Shot 2022-03-02 at 10 27 44 PM](https://user-images.githubusercontent.com/98017/156508946-9ad482ab-f51d-44ce-9d79-a40fee68a89f.png)

Reply:
Good callout, it can be a Job+launch instead of Deferred+async now after all the other refactoring.